io/io/src/TBufferFile.cxx

```diff
@@ -1328,16 +1492,31 @@ void TBufferFile::ReadFastArray(Short_t *h, Int_t n)
    if (n <= 0 || l > fBufSize) return;
 
 #ifdef R__BYTESWAP
+   if (IsBufBigEndian()) {
 # ifdef USE_BSWAPCPY
-   bswapcpy16(h, fBufCur, n);
-   fBufCur += sizeof(Short_t)*n;
+      bswapcpy16(h, fBufCur, n);
```

Comment:
Because I added 'if' statement there, and the dark green indicates three white spaces.